pkg/spdk/engine.go

```diff
@@ -934,20 +936,11 @@ func (e *Engine) ReplicaAdd(spdkClient *spdkclient.Client, dstReplicaName, dstRe
 	if err != nil {
 		return err
 	}
-	defer func() {
-		if err != nil {
-			srcReplicaServiceCli.Close()
-		}
-	}()
 	dstReplicaServiceCli, err := GetServiceClient(dstReplicaAddress)
 	if err != nil {
+		srcReplicaServiceCli.Close()
```

Comment:
Handled in https://github.com/longhorn/longhorn-spdk-engine/pull/203/files `getSrcAndDstReplicaClients`